### ISSUE TITLE
Experiment with 'Pay What You Want' proposals / initiatives 

### ISSUE THREAD

**Issue by monsieurbulb** [virto-network/kreivo#425]
In general people agree that common good treasuries should spend on something - so far so good.

Sadly, beyond that it’s a free for all of subjective opinion.

Most contentious of all is the funding amount - if you want to trigger people, just set your request to anything above $10k and watch the treasury guardians implode.

- How much should an event cost?
- What is your code worth?
- Are we underpaying or overpaying for media?

Solution 

> Pay what you want (or PWYW, also referred to as value-for-value model [[1]](https://en.wikipedia.org/wiki/Pay_what_you_want#cite_note-1)[[2]](https://en.wikipedia.org/wiki/Pay_what_you_want#cite_note-2)) is a [pricing strategy](https://en.wikipedia.org/wiki/Pricing_strategies) where buyers pay their desired amount for a given commodity. This amount can sometimes include zero. A minimum (floor) price may be set, and/or a suggested price may be indicated as guidance for the buyer. The buyer can select an amount higher or lower than the standard price for the commodity.[[3]](https://en.wikipedia.org/wiki/Pay_what_you_want#cite_note-pay2-3)[[4]](https://en.wikipedia.org/wiki/Pay_what_you_want#cite_note-Smart_Pricing2-4) Many common PWYW models set the price prior to a purchase (ex ante), but some defer price-setting until after the experience of consumption (ex post) (similar to [tipping](https://en.wikipedia.org/wiki/Gratuity)). PWYW is a buyer-centered form of participatory pricing, also referred to as co-pricing (as an aspect of the [co-creation](https://en.wikipedia.org/wiki/Co-creation) of value). - [Wikipedia](https://en.wikipedia.org/wiki/Pay_what_you_want)

The aim is to hack together an approach where we merge elements of Gov1 - amount set by median of council tips, with Gov2’s open tracks and referendums.

> A group of Tippers is determined through the config Config. After half of these have declared some amount that they believe a particular reported reason deserves, then a countdown period is entered where any remaining members can declare their tip amounts also.

> After the close of the countdown period, the median of all declared tips is paid to the reported beneficiary, along with any finders fee, in case of a public (and bonded) original report.

Have already explored with Ideal Network whose new [Kusama pallet + bridge](https://kusama.polkassembly.io/post/2645) enables some of this stuff via timelock encryption.

> Timelock Encryption: drand's output enables us to build timelock encryption capabilities on top of the beacon, enabling new kinds of use cases for trustless protocols, such as trustless asset swaps, [keyless crypto wallets](https://forum.polkadot.network/t/a-keyless-wallet-concept-using-timelock-encryption/7096), and more.

See 
https://forum.polkadot.network/t/kusama-rfc-pay-what-you-want/3709
https://ideallabs.substack.com/p/p2p-semi-autonomous-games-on-the-978

**Comment by driemworks** [comment]
A prototype of something like that would be fairly straightforward to build imo. It's not far off conceptually from some contracts we've already built https://github.com/ideal-lab5/contracts/blob/main/examples/transmutation/lib.rs

**Comment by olanod** [comment]
This also seems to relate to the [prize discovery topic](https://forum.polkadot.network/t/a-price-discovering-treasury-proposal/9083?u=olanod) Shawn Tabrizi started in the forum. I'd be interested to integrate something like that with pallet-referenda and allow special kinds of proposals meant to discover a value(a prize in this case) to then be used as input of a regular payment proposal that considers a "secret minimum amount" a proponent previously submitted.

**Comment by monsieurbulb** [comment]
@olanod yup. shared concept a while back here https://forum.polkadot.network/t/kusama-rfc-pay-what-you-want/3709

would be a fun experiment and as @driemworks says could be very quick to implement an interesting experiment.

**Comment by driemworks** [comment]
I elaborated on some further specifics here, with some inspiration from Shawn's usage of the median (rather than average) in the forum post above: https://hackmd.io/@Y5vcBYL4SyeRG_CqQq0DoQ/rypzvOHq0 .

**Comment by olanod** [comment]
Closing as part of the Kreivo v2 clean slate. Relevant scope will be tracked with new issues under the [Kreivo v2 milestone](https://github.com/virto-network/kreivo/milestone/10).
